pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java

```diff
@@ -399,64 +487,51 @@ private static String generateConstructor(
 					if (shouldThrowOnOneOfNull && field instanceof OneOfField) {
 						sb.append(generateConstructorCodeForField(field)).append('\n');
 					}
-					switch (field.type()) {
-						case BYTES, STRING: {
-							sb.append("this.$name = $name != null ? $name : $default;"
-									.replace("$name", field.nameCamelFirstLower())
-									.replace("$default", getDefaultValue(field, msgDef, lookupHelper))
-							);
-							break;
-						}
-						case MAP: {
-							sb.append("this.$name = PbjMap.of($name);"
-									.replace("$name", field.nameCamelFirstLower())
-							);
-							break;
-						}
-						default:
-							if (field.repeated()) {
-								sb.append("this.$name = $name == null ? Collections.emptyList() : $name;".replace("$name", field.nameCamelFirstLower()));
-							} else {
-								sb.append("this.$name = $name;".replace("$name", field.nameCamelFirstLower()));
+					if (field.nameCamelFirstLower().equals("precomputedHashCode")) {
+						final String computeHashCode = "computeHashCode("
+								+ fields.stream()
+								.filter(f -> f.fieldNumber() != -1)
+								.map(f -> "this."+f.nameCamelFirstLower())
+								.collect(Collectors.joining(", "))
+								+ ")";
+						sb.append("this.precomputedHashCode = "+computeHashCode+";");
+					} else if (field.nameCamelFirstLower().equals("protobufEncodedSize")) {
+						final String computeProtobufSize = "computeProtobufSize("
+								+ fields.stream()
+								.filter(f -> f.fieldNumber() != -1)
+								.map(f -> "this."+f.nameCamelFirstLower())
+								.collect(Collectors.joining(", "))
+								+ ")";
+						sb.append("this.protobufEncodedSize = "+computeProtobufSize+";");
```

Comment:
I wonder if it should be done lazily rather than in the constructor. Are all protobuf objects measured sooner or later?

Reply:
Ah, I see, these are Java records, so all fields must be initialized in the constructor and never changed in the future...

Reply:
Yep, trying changing to class